### DISCUSSION TITLE
viseron monitoring

### DISCUSSION THREAD

**Discussion by k0stett** [roflcoopter/viseron#547] [Ideas]
I like to link with Zabbix all services running in my homenet, so lot of them have HTTP API methods with metrics like resources consumption or error statements. I am getting some metrics from MQTT (camera connection state, 4ex) but it's not enough for full understanding of viseron state, and today I've found that nvenc got failed at night and all recordings is zero-sized but `docker logs viseron` was the only place with errors. It would be better to have these errors in place which is more compat with monitoring systems. Thanks.

**Comment by roflcoopter** [comment]
I see, did you find the error in the log as an uncaught exception?

Basically what you want is something like the [Sentry integration in Home Assistant](https://www.home-assistant.io/integrations/sentry/) but for Zabbix, right?

**Reply by k0stett** [reply]
> I see, did you find the error in the log as an uncaught exception?

Yep, only in viseron container log without dot-separated path to module. I can catch these errors with regex logs watching but if an error doesn't match my regexes I'll miss that error.

> Basically what you want is something like the [Sentry integration in Home Assistant](https://www.home-assistant.io/integrations/sentry/) but for Zabbix, right?

So maybe (I was never using Sentry), more like Proxmox HTTP API in its Audit part. Or Sentry for viseron but not HASS :)

**Reply by roflcoopter** [reply]
Would sentry be enough?
Sentry has great support for python https://docs.sentry.io/platforms/python/ while zabbix seems to be less used.

Sentry would probably be really simple to add while zabbix requires a lot more work
